ProcessLib/Process.h

```diff
@@ -72,14 +72,59 @@ class Process
 	/// Process specific initialization called by initialize().
 	virtual void createLocalAssemblers() = 0;
 
-	/// Postprocessing after solve().
+	/// Preprocessing before starting assembly for new timestep.
+	virtual void preTimestep(GlobalVector const& x) {}
```

Comment:
Will `preTimestep()`, `postTimestep()` or `output()` change the internals of the process object or will they work on external data? If they only work on external data the methods could be `const`.